lib/classes/compatibility/lite-speed-cache.php

```diff
@@ -54,6 +54,51 @@ public function module_init($sm) {
         add_action('sm:pre::synced::image', array($this, 'update_md5_and_manual_sync'), 10, 1);
       }
 
+      /**
+       * Get the md5_file hash from cloud meta.
+       * 
+       * @param int $attachment_id
+       * @return array
+       */
+      private function _get_md5_meta($attachment_id) {
+        // If WP-Stateless version >= 4.0.0
+        if ( function_exists('ud_stateless_db') ) {
+          $cloud_meta = [
+            'fileMd5' => ud_stateless_db()->get_file_meta_value($attachment_id, 'fileMd5', []),
+          ];
+        } else {
+          $cloud_meta = get_post_meta($attachment_id, 'sm_cloud', true);
+        }
+
+        return $cloud_meta;
+      }
+
+      /**
+       * Update the md5_file hash in cloud meta.
+       * 
+       * @param int $attachment_id
+       * @param array $cloud_meta
+       */
+      private function _update_md5_meta($attachment_id, $cloud_meta) {
+        // If WP-Stateless version >= 4.0.0, postmeta can be limited to 'fileMd5' only
+        // and we should not override it 
+        $test_meta = $cloud_meta;
+        unset($test_meta['fileMd5']);
+        
+        if ( empty($test_meta) ) {
+          $test_meta = get_post_meta($attachment_id, 'sm_cloud', true);
+          $test_meta['fileMd5'] = $cloud_meta['fileMd5'];
+          $cloud_meta = $test_meta;
+        }
+
+        update_post_meta($attachment_id, 'sm_cloud', $cloud_meta);
```

Comment:
@balexey88 should we do this if function `ud_stateless_db` exists?
I think we can skip it to prevent duplicating

Reply:
This is a little bit tricky moment. The same code changes are reflected in LiteSpeed Cache Addon plugin. And this keeping the code compliant with v4.0.0, v3.4.0, and below.

If `ud_stateless_db` does not exists or `! WP_STATELESS_POSTMETA` - then `$cloud_meta` input parameter will contain all the meta data needed for `update_post_meta(...`  (from `get_post_meta`)
Otherwise input parameter `$cloud_meta` will contain only `['fileMd5']` field, because other data is not really needed for the data processing (from `wp_stateless_file_meta` table).

So as long as we keep `sm_cloud` meta in sync with our custom tables, this _crutch_ is needed, and it is only needed here.